tests/components/ring/test_switch.py

```diff
@@ -0,0 +1,86 @@
+"""The tests for the Ring switch platform."""
+from asynctest import patch
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from tests.common import load_fixture
+from .common import setup_platform
+
+
+async def test_entity_registry(hass, requests_mock):
+    """Tests that the devices are registed in the entity registry."""
+    with patch("ring_doorbell._exists_cache", return_value=False):
+        with patch("ring_doorbell._save_cache", return_value=True):
+            await setup_platform(hass, SWITCH_DOMAIN)
+            entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+            entry = entity_registry.async_get("switch.front_siren")
+            assert entry.unique_id == "aacdef123-siren"
+
+            entry = entity_registry.async_get("switch.internal_siren")
+            assert entry.unique_id == "aacdef124-siren"
+
+
+async def test_siren_off_reports_correctly(hass, requests_mock):
+    """Tests that the initial state of a device that should be off is correct."""
+    with patch("ring_doorbell._exists_cache", return_value=False):
+        with patch("ring_doorbell._save_cache", return_value=True):
+            await setup_platform(hass, SWITCH_DOMAIN)
+
+            state = hass.states.get("switch.front_siren")
```

Comment:
Getting the state doesn't need to be done within the context manager.

Reply:
How else can this be done? Is it an issue with it being done this way?

Reply:
Yes, flat is better than nested.